lib/msal-browser/src/interaction_handler/RedirectHandler.ts

```diff
@@ -3,31 +3,28 @@
  * Licensed under the MIT License.
  */
 
-import { AuthorizationCodeClient, StringUtils, AuthorizationCodeRequest, ICrypto, AuthenticationResult, ThrottlingUtils } from "@azure/msal-common";
+import { AuthorizationCodeClient, StringUtils, AuthorizationCodeRequest, ICrypto, AuthenticationResult, ThrottlingUtils, Authority, INetworkModule } from "@azure/msal-common";
 import { BrowserAuthError } from "../error/BrowserAuthError";
 import { BrowserConstants, TemporaryCacheKeys } from "../utils/BrowserConstants";
 import { BrowserUtils } from "../utils/BrowserUtils";
 import { BrowserProtocolUtils } from "../utils/BrowserProtocolUtils";
 import { BrowserCacheManager } from "../cache/BrowserCacheManager";
+import { InteractionHandler } from "./InteractionHandler";
 
-export class RedirectHandler {
+export class RedirectHandler extends InteractionHandler {
 
-    private authModule: AuthorizationCodeClient;
-    private browserStorage: BrowserCacheManager;
     private browserCrypto: ICrypto;
-    private authCodeRequest: AuthorizationCodeRequest;
 
     constructor(authCodeModule: AuthorizationCodeClient, storageImpl: BrowserCacheManager, browserCrypto: ICrypto) {
-        this.authModule = authCodeModule;
-        this.browserStorage = storageImpl;
+        super(authCodeModule, storageImpl);
         this.browserCrypto = browserCrypto;
     }
 
     /**
      * Redirects window to given URL.
      * @param urlNavigate
      */
-    initiateAuthRequest(requestUrl: string, authCodeRequest: AuthorizationCodeRequest, redirectTimeout: number, redirectStartPage?: string): Promise<void> {
+    initiateAuthRequest(requestUrl: string, authCodeRequest: AuthorizationCodeRequest, redirectTimeout?: number, redirectStartPage?: string): Promise<void> {
```

Comment:
Why is `redirectTimeout` set to optional? FYI in #2669, I also made `redirectStartPage` required, since it will always be passed: https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/2669/files#diff-d592fb8a2a52b3d36d8625f2210c27b8360c6d1f0cff4fe06494d1baf77ac593

Reply:
Oh, is it because it now implements `InteractionHandler`?

Reply:
Yeah that's correct. The interface implementation does not include redirectTimeout.

Reply:
Can we remove the `initiateAuthRequest` abstract from `InteractionHandler` so that we dont have to make these optional?

Reply:
I would like to keep the abstract function as a constraint for subclasses implementing this. I understand that there needs to be a compiler level way to ensure that the redirectTimeout and redirectStartPage are passed, but for now can we ensure this by adding code errors?

Reply:
I personally don't like making a param optional but then throwing if it's not provided. Would prefer either making this its own function that doesn't implement the abstract or getting rid of the abstract altogether.

Reply:
I personally think it's worse design to take out the abstract functionality than to check if an optional parameter is provided or not. Let me see if I can find an alternative, but in the meantime can you approve the PR so I can merge as soon as it is ready?

Reply:
I also don't like the idea of throwing an error if the params are not passed. I would rather the code just handle it gracefully.

Reply:
So you can also just leave this as you have it, and I can deal with adjustments that need to be made for my PR.

Reply:
made an update to the `initiateAuthRequest` function, please check and see if it works for you guys.